bytecomp/bytelink.ml

```diff
@@ -344,7 +344,7 @@ let link_bytecode ?final_name tolink exec_name standalone =
          let runtime = make_absolute !Clflags.use_runtime in
          let runtime =
            (* shebang mustn't exceed 128 including the #! and \0 *)
-           if String.length runtime > 125 then
+           if String.length runtime > 125 || String.contains runtime ' ' then
              "/bin/sh\n\
               exec \"" ^ runtime ^ "\" \"$0\" \"$@\""
```

Comment:
While you're here, this small change from https://github.com/dra27/ocaml/commit/bec4f8dbc3c79f8a1d4d7f7abcdd14965ecd3e4d which fixes https://github.com/ocaml/ocaml/pull/8622#discussion_r328158394 could also be dropped in:
```suggestion
              exec " ^ Filename.quote runtime ^ " \"$0\" \"$@\""
```